src/leap/common/events/client.py

```diff
@@ -55,275 +54,269 @@ class CallbackAlreadyRegistered(Exception):
-    :type signal: int (see the `events.proto` file)
-    :param content: the contents of the event signal
-    :type content: str
-    :param mac_method: the method used for auth mac
-    :type mac_method: str
-    :param mac: the content of the auth mac
-    :type mac: str
-    :param reqcbk: a callback to be called when a response from server is
-                   received
-    :type reqcbk: function(proto.SignalRequest, proto.EventResponse)
-    :param timeout: the timeout for synch calls
-    :type timeout: int
-
-    :return: the response from server for synch calls or nothing for asynch
-             calls.
-    :rtype: leap.common.events.events_pb2.EventsResponse or None
-    """
-    request = proto.SignalRequest()
-    request.event = signal
-    request.content = content
-    request.mac_method = mac_method
-    request.mac = mac
-    service = RpcService(proto.EventsServerService_Stub, server.SERVER_PORT,
-                         'localhost')
-    logger.debug("Sending signal to server: %s", str(request)[:40])
-    return service.signal(request, callback=reqcbk, timeout=timeout)
-
-
-def ping(port, reqcbk=None, timeout=1000):
-    """
-    Ping a client running in C{port}.
-
-    :param port: the port in which the client should be listening
-    :type port: int
-    :param reqcbk: a callback to be called when a response from client is
-                   received
-    :type reqcbk: function(proto.PingRequest, proto.EventResponse)
-    :param timeout: the timeout for synch calls
-    :type timeout: int
-
-    :return: the response from client for synch calls or nothing for asynch
-             calls.
-    :rtype: leap.common.events.events_pb2.EventsResponse or None
-    """
-    request = proto.PingRequest()
-    service = RpcService(
-        proto.EventsClientService_Stub,
-        port,
-        'localhost')
-    logger.debug("Pinging a client in port %d..." % port)
-    return service.ping(request, callback=reqcbk, timeout=timeout)
+    def _init_zmq(self):
+        """
+        Initialize ZMQ connections.
+        """
+        self._context = zmq.Context()
+        self._loop = EventsIOLoop()
+        self._push = self._zmq_connect_push()
+        self._sub = self._zmq_connect_sub()
 
+    def _zmq_connect(self, socktype, port):
+        """
+        Initialize one ZMQ connection.
 
-class EventsClientService(proto.EventsClientService):
-    """
-    Service for receiving signal events in clients.
-    """
+        :param socktype: The ZMQ socket type.
+        :type socktype: int
+        :param port: The TCP port to connect to.
+        :type port: int
 
-    def __init__(self):
-        proto.EventsClientService.__init__(self)
+        :return: A ZMQ connection stream.
+        :rtype: ZMQStream
+        """
+        socket = self._context.socket(socktype)
+        stream = zmqstream.ZMQStream(socket, self._loop)
+        socket.connect("tcp://127.0.0.1:%d" % port)
```

Comment:
should we use CurveZMQ here and fallback to ipc with socket permissions as we did on the client?